### PR TITLE
Add django-helusers integration

### PR DESCRIPTION
**Pull request by mingfeng**
This PR adds `django-helusers` integration to the project. Since we are switching the user model, some manual cleanup in database is needed  (confirmed with project manager that it's OK to clean up those data). Following are the steps that needs to be carried out to switch the user model:

- reset the migrations for auth, admin, files and imports apps by running the commands below. This will remove all the data in the models of these three apps.
```
# this will reset all migrations that depend on user model, inlcuding admin, files and imports
python manage.py run migrate auth zero 
```
- deploy new code
- re-apply the migrations for the above three apps and the new user app (should be automatically run when deploying new code)
- re-create users in the new user model 

Refs: #237

### CODE REVIEW

File: ltj/urls.py [hunk 1]
```diff
@@ -1,30 +1,15 @@
...
-    url(r"v1/", include(router.urls)),
-    url(r"^ltj/", include("nature.urls")),
+    path("helauth/", include("helusers.urls")),
+    path("admin/", admin.site.urls),
+    path("v1/", include(router.urls)),
+    path("ltj/", include("nature.urls")),
 ]
 
 admin.site.site_title = _("Luontotietojärjestelmä site admin")
-admin.site.site_header = _("Luontotietojärjestelmä administration")
```
Comment (frwickst) [inline ltj/urls.py:29]: Why was this line removed?
Not questioning the removal, probably a good point behind it, just wondering since this commits does not mention it.

File: users/tests.py [hunk 2]
```diff
@@ -1,3 +0,0 @@
-from django.test import TestCase
```
Comment (frwickst) [inline users/tests.py]: Would it make sense to remove these files already in the initial commit that introduced the `user` app to minimize code churn?

File: users/views.py [hunk 3]
```diff
@@ -1,3 +0,0 @@
-from django.shortcuts import render
```
Comment (frwickst) [inline users/views.py]: Same with this file.

**Review by frwickst — CHANGES_REQUESTED** [review]
Looks good overall and it seems like you have done the research into what needs to happen in order to migrate to a custom user model.

The only thing that could be cleaned up is the slight code churn regarding the new user model.

The following are mainly sanity checks, but I'll ask them anyway so that things are double-checked :)
- Does this remove all of the current users in the system? Asking as if not, they need to be migrated, but I guess that the zeroing the `admin` migrations and deleting that data will also delete all users, right?
- If there are any relations to the user model, which I imagine that there are, how are those handled? Are the only relations currently found in the `files` and `imports` apps?

Good work @mingfeng 👍

File: users/tests.py (hunk 2 above)
Comment (mingfeng) [reply users/tests.py]: Good point, I think we can combine the two commits that create the app and add the user model integration.

File: ltj/urls.py (hunk 1 above)
Comment (mingfeng) [reply ltj/urls.py:29]: The reason for that is because `django-helusers` override the AdminSite class and make this a read-only property, which prevents us assigning a new header to it. I think I'll make a PR to the `django-helusers` project to make it possible to override the property.

**Comment by mingfeng** [comment]
@frwickst There are no relations in other apps other than `files`, `imports` and `admin`. I think the user data are not cleared because they are in the auth app, but I think we should also reset the migrations for the auth app.

**Review by frwickst — APPROVED** [review]
LGTM
